### DISCUSSION TITLE
ScriptReference/EditorUserBuildSettings-activeScriptCompilationDefines

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#194] [Notes]
# ScriptReference/EditorUserBuildSettings-activeScriptCompilationDefines

https://docs.unity3d.com/ScriptReference/EditorUserBuildSettings-activeScriptCompilationDefines.html

**Comment by mvi** [comment]
activeScriptCompilationDefines includes:  
- Internal defines set by Unity such as `UNITY_5_6_OR_NEWER`  
- User defines set through Player Settings' Scripting Define Symbols, common ones include `CROSS_PLATFORM_INPUT` (from Sample Assets) and `TMP_PRESENT` (if you have TextMeshPro)  
- C# defines such as `DEBUG` and `TRACE`  
Unfortunately it doesn't seem to include defines set in [rsp files](https://docs.unity3d.com/Manual/PlatformDependentCompilation.html) such as `mcs.rsp` and `csc.rsp`
